### ISSUE TITLE
Which modules should I disable for minimal functionality in Windows 7?

### ISSUE THREAD

**Issue by shiyuwang-jamk** [snolab/CapsLockX#74]
I cloned the CLX repo back on the 3rd of April to a Windows 7 host with no internet access. The programs in the host are limited, so for speed and simplicity, I tried to modify `CapsLockX-Config.ini` to disable most modules and keep only the ones that control the mouse cursor and other basic, stock functionality. And then I ran into errors coming out of a particular conditional code in an AHK script, such as Line 164 of the `WindowManager` module.

How do I figure out the minimal setup? Do you have any recommendations for legacy platforms like my host running Windows 7?

**Comment by snomiao** [comment]
just delete the fioes u dont need

2023年4月25日(火) 14:02 Wang Dani ***@***.***>:

**Comment by shiyuwang-jamk** [comment]
What are fioes?
________________________________

**Comment by snomiao** [comment]
files

**Comment by shiyuwang-jamk** [comment]
Here is what I have found so far:
Of all the modules available at CLX, I have found the following two modules would cause a fatal halt if disabled:
- Blender
- Virtual Desktop

The prompt when faulted looks like this:
`Error in #include file
"..\..\Modules\CLX-WindowManager.ahk":
  Call to nonexistent function

Specifically: IsWindowOnCurrentVIrtualDesktop(hWnd))`

and pointing at line 164 of the module AHK file.

Similar issue with Blender disabled, pointing at line 38 containing conditional `Blender窗口内() && Blender增强模式` in the `.\Modules\QuickTips.ahk`

You suggested deleting certain files/modules, and so I did. And yet, this could result in a slippery slope, for instance, where the `QuickTips.ahk` relies on the `WindowManager`, such as the `任务栏中()` conditional at line 32 of `QuickTips`

So it turned out that deleting certain modules won't work and causes even more headaches. Seems I will have to make my own version of `QuickTips` as part of the minimization effort.
